script/tool/lib/src/pubspec_check_command.dart

```diff
@@ -76,6 +100,40 @@ class PubspecCheckCommand extends PackageLoopingCommand {
   PackageLoopingType get packageLoopingType =>
       PackageLoopingType.includeAllSubpackages;
 
+  @override
+  Future<void> initializeRun() async {
+    // Find all local, published packages.
+    await for (final File pubspecFile in packagesDir.parent
+        .list(recursive: true, followLinks: false)
+        .where((FileSystemEntity entity) =>
+            entity is File && p.basename(entity.path) == 'pubspec.yaml')
+        .map((FileSystemEntity file) => file as File)) {
```

Comment:
You could use `whereType()`, up to you though.
```suggestion
    for (final File pubspecFile in (await packagesDir.parent
        .list(recursive: true, followLinks: false)
        .toList())
        .whereType<File>()
        .where((File entity) => p.basename(entity.path) == 'pubspec.yaml')) {
```